e2e/openssl/openssl_test.go

```diff
@@ -62,6 +61,28 @@ func TestOpenSSL(t *testing.T) {
 	ct.Init(t, resources)
 	require.True(t, t.Run("generate", ct.Generate), "contrast generate needs to succeed for subsequent tests")
 
+	if platform == platforms.K3sQEMUSNP {
+		// The generate command doesn't fill in all required fields when
+		// generating a manifest for baremetal SNP. Do that now.
+
+		manifestBytes, err := os.ReadFile(ct.WorkDir + "/manifest.json")
+		require.NoError(t, err)
+		var m manifest.Manifest
+		require.NoError(t, json.Unmarshal(manifestBytes, &m))
+
+		for i, snp := range m.ReferenceValues.SNP {
+			snp.MinimumTCB.BootloaderVersion = toPtr(manifest.SVN(0))
+			snp.MinimumTCB.TEEVersion = toPtr(manifest.SVN(0))
+			snp.MinimumTCB.SNPVersion = toPtr(manifest.SVN(0))
+			snp.MinimumTCB.MicrocodeVersion = toPtr(manifest.SVN(0))
+			m.ReferenceValues.SNP[i] = snp
+		}
+
+		manifestBytes, err = json.Marshal(m)
+		require.NoError(t, err)
+		require.NoError(t, os.WriteFile(ct.WorkDir+"/manifest.json", manifestBytes, 0o644))
+	}
+
```

Comment:
Not a fan of doing this in the test. Adding per-platform workarounds here make the code hard to maintain. I'd favor to fill in the missing values in the command-line (probably hard due to how the test framework works), or factor this out into its own function.